x/poolmanager/README.md

```diff
@@ -508,31 +511,24 @@ type TakerFeeDistributionPercentage struct {
-
-For community pool, this is just a direct send to community pool.
+We search for a KVStore taker fee override before utilizing the default taker fee. Just as before (order does not matter), a KVStore entry for OSMO<>USDC exists, so we utilize a 0.01% taker fee instead of the 0.02% default taker fee. 0.01% of 1000 OSMO is 1 OSMO.
 
-For staking, we actually ALSO send this to the `non_native_fee_collector`. At epoch time, this OSMO is just skipped over, while everything else is swapped to OSMO. At the very end, it takes the OSMO directly sent to the `non_native_fee_collector` along with the non native tokens that were just swapped to OSMO and distributes it to stakers.
+At time of swap, all 1 OSMO is sent to the `taker_fee_collector` module account. Again, nothing is done with any taker fee funds until epoch.
 
-### Important Note: How to extract the data
+At epoch, we check the `OsmoTakerFeeDistribution`. In this example, let’s say its 20% to community pool and 80% to stakers. This means that 0.2 OSMO is set for community pool and 0.8 is set for stakers.
 
-If one were to take the total amount of tokens in the two module accounts (`non_native_fee_collector` and `non_native_fee_collector_community_pool`), this would be slightly over exaggerating the amount that is generated from taker fees. This is because, when a user uses a non-native token as a FEE TOKEN, this is also sent to the `non_native_fee_collector`. So there are two options here to extract the info:
+For community pool, this is just a direct send of the OSMO to the community pool.
 
-1. Track the delta of the module accounts 1 block after epoch X and 1 block before epoch X+1. Also, track the total non osmo txfees generated in this period. Subtract the total non osmo txfees generated in this period from the delta of the `non_native_fee_collector`. Add this to the delta of `non_native_fee_collector_community_pool` values. This is the taker fees generated
-2. This is the less better way, but you can track the `SendCoinsFromAccountToModule` events from each block. The problem with this is, imagine I swap USDC to OSMO and use USDC as txfee. This would generate three `SendCoinsFromAccountToModule` events:
-   1. Txfee gets sent to `non_native_fee_collector`
-   2. Part of taker fee gets sent to `non_native_fee_collector`
-   3. Other part of taker fee gets sent to `non_native_fee_collector_community_pool`
-   You could make an assumption here that the txfee is going to be the smaller of the two that gets sent to the `non_native_fee_collector`, or better the order of operations is going to always be the same, so you can figure out if the first or second send to `non_native_fee_collector` is the txfee and not track that value
\ No newline at end of file
+For staking, the OSMO is directly sent to the auth module account, which distributes it to stakers.
```

Comment:
```suggestion
For staking, the OSMO is directly sent to the auth "fee collector" module account, which distributes it to stakers.
```

Reply:
Note to self, we want epochs to come before staking in begin block ordering

Reply:
Ok I checked that it is before staking. Could you add a comment here, just reminding ourselves, that epochs must come before staking, due to txfees needs: https://github.com/osmosis-labs/osmosis/blob/dev/protorev_no_takerfee_est/app/modules.go#L203

Reply:
Comment added here https://github.com/osmosis-labs/osmosis/pull/7555/commits/cf51e75db231575dfe804d910dc13cf13115013f